src/apps/chifra/proto/chifra.proto

```diff
@@ -0,0 +1,55 @@
+syntax = "proto3";
+option go_package = "github.com/TrueBlocks/trueblocks-core/src/apps/chifra/proto";
+
+service Names {
+    // Search
+    rpc Search(SearchRequest) returns (SearchResponse) {}
+    rpc SearchStream(SearchRequest) returns (stream Name) {}
+    // CRUD
+    rpc Create(CreateRequest) returns (CRUDResponse) {}
+    rpc Update(CreateRequest) returns (CRUDResponse) {}
+    rpc Delete(DeleteRequest) returns (CRUDResponse) {}
+    rpc Undelete(DeleteRequest) returns (CRUDResponse) {}
+    rpc Remove(DeleteRequest) returns (CRUDResponse) {}
+}
+
+message SearchRequest {
+    int64 parts = 1;
+    repeated string terms = 2;
+    int64 sort = 3;
+}
+
+message SearchResponse {
+    repeated Name names = 1;
+}
+
+message Name {
+    string address = 1;
+    optional uint64 decimals = 2;
+    optional bool deleted = 3;
+    optional bool isContract = 4;
+    optional bool isCustom = 5;
+    optional bool isErc20 = 6;
+    optional bool isErc721 = 7;
+    optional bool isPrefund = 8;
+    string name = 9;
+    optional string petname = 10;
+    optional string source = 11;
+    optional string symbol = 12;
+    optional string tags = 13;
```

Comment:
I'm not quite sure what `optional` means here, but I do know that `tags` cannot be empty. I think `clean` (which should be called by all the `create` and `edit` crud routines should default it to `30-Contracts` if it's a contract and something else if it's not a contract. I'm pretty sure we have tests for that, but if not, we should.

Reply:
Optional means that you don't have to set this field. I can make `tags` required, but from what you write it's optional, because even if it's missing, it gets set by `clean`.